modules/claims_api/app/sidekiq/claims_api/reporting_base.rb

```diff
@@ -142,22 +142,18 @@ def errored_ews
     end
 
     def map_transaction_ids(array)
-      # Dynamically generate unique keys like A, B, C, etc.
       transaction_mapping = {}
-      key_sequence = ('A'..'Z').to_a
-      key_index = 0
 
       # Map each unique transaction_id to a new key
-      array.each do |item|
+      array&.each do |item|
         transaction_id = item[:transaction_id]
-        unless transaction_mapping.key?(transaction_id)
-          transaction_mapping[transaction_id] = key_sequence[key_index]
-          key_index += 1
+        if !transaction_mapping.key?(transaction_id) && transaction_id.present?
+          transaction_mapping[transaction_id] = transaction_id[-5..transaction_id.length]
         end
       end
 
       # Group the array by the new keys
-      array.group_by { |item| transaction_mapping[item[:transaction_id]] }
+      array.group_by { |item| transaction_mapping[item[:transaction_id]] }.sort
```

Comment:
Could we group the transactions by an auto-incrementing integer?

```rb
def map_transaction_ids(transactions)
  transaction_id_to_key = Hash.new { |hash, transaction_id| hash[transaction_id] = hash.size + 1 }

  # Group transactions by their assigned unique integer keys
  transactions&.group_by { |transaction| transaction_id_to_key[transaction[:transaction_id]] }.sort
end
```

or is there a reason we are using the last five?

Reply:
we can use the erb to do the heavy lifting for the incrementation as well. we already have the each_with_index baked in, and we aren't currently utilizing that index. 

when I was dabbling with this after the first night returned 0 errored submissions I moved to something along the lines of 
```
{}.tap do |ob|
  array.map do |elem|
    ob[elem[0]] = [] unless ob[elem[0]].present?
    ob[elem[0]] << elem[1]
  end
end
```
which sets the object key to the transaction ID and then pushes all LH id's to said corresponding array when found.

if you want it to be a bit more readable, pretty sure you can reach the destructured objects here, so you could do something like 
```
ob[elem[:transaction_id] = [] unless ob[elem[:transaction_id]].present?
ob[elem[:transaction_id]] << elem[:id]
```

probably 1,000,000 ways to do this though.